src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -68,12 +76,17 @@ const CustomFieldInputDate = forwardRef(function CustomFieldInputDate(props, ref
 
   useImperativeHandle(ref, () => ({
     get dirty() {
-      return props.value !== this.value;
+      return convertToFormat(props.value, 'yyyy-mm-dd') !== this.value;
     },
     id: props.id,
     name: props.name,
     get value() {
```

Comment:
I think getting spec coverage with the new functionality will be useful. We want to make sure that the value is serializable and consistent regardless of the state of the component.